distsql/distsql.go

```diff
@@ -141,3 +144,74 @@ func Checksum(ctx context.Context, client kv.Client, kvReq *kv.Request, vars *kv
 	}
 	return result, nil
 }
+
+// SetEncodeType sets the encoding method for the DAGRequest. Possible encoding
+// methods are:
+// 1. arrow: result is encoded using the apache arrow format.
+// 2. default: result is encoded row by row.
+func SetEncodeType(dagReq *tipb.DAGRequest) {
+	if useChunkIPC(dagReq) {
+		dagReq.EncodeType = tipb.EncodeType_TypeArrow
+	} else {
+		dagReq.EncodeType = tipb.EncodeType_TypeDefault
+	}
+}
+
+func useChunkIPC(dagReq *tipb.DAGRequest) bool {
+	if config.GetGlobalConfig().TiKVClient.EnableArrow == false {
+		return false
+	}
+	if config.GetGlobalConfig().EnableStreaming == true {
+		return false
+	}
+	root := dagReq.Executors[len(dagReq.Executors)-1]
```

Comment:
The checks for root is not necessary, cause it has been checked before SetEncodeType